pkg/composer/composer.go

```diff
@@ -123,18 +133,88 @@ func (c *Composer) runNerdctlCmd(ctx context.Context, args ...string) error {
 	return nil
 }
 
-func findComposeYAML() (string, error) {
-	yamlNames := []string{"docker-compose.yml", "docker-compose.yaml", "compose.yml", "compose.yaml"}
-	for _, candidate := range yamlNames {
-		fullPath, err := filepath.Abs(candidate)
+//find compose Yaml file in current directory and its parents
+func findComposeYAML(o *Options) (string, error) {
+	pwd, err := o.GetWorkingDir()
+	if err != nil {
+		return "", err
+	}
+	for {
+		yamlNames := []string{"docker-compose.yml", "docker-compose.yaml", "compose.yml", "compose.yaml"}
+		for _, candidate := range yamlNames {
+			f := filepath.Join(pwd, candidate)
+			if _, err := os.Stat(f); err == nil {
+				return f, nil
+			} else if !os.IsNotExist(err) {
+				return "", err
+			}
+		}
+		parent := filepath.Dir(pwd)
+		if parent == pwd {
+			return "", errors.Errorf("cannot find a compose YAML, supported file names: %+v in this directory or any parent", yamlNames)
+		}
+		pwd = parent
+	}
+}
+
+// setDotEnv was inspired from https://github.com/compose-spec/compose-go/blob/de56f4f0cb3c925f41df594221148613534c2cd3/cli/options.go#L174-215
```

Comment:
Why not use the conpose-go code directly?
What is the difference?

Reply:
If something is missing in the compose-go, please consider contributing to the compose-go upstream

Reply:
Small nits, PR opened https://github.com/compose-spec/compose-go/pull/166